src/MongoDB.EntityFrameworkCore/Serializers/EntitySerializer.cs

```diff
@@ -66,7 +66,9 @@ public bool TryGetMemberSerializationInfo(string memberName, out BsonSerializati
             if (navigation != null)
             {
                 var entityType = navigation.TargetEntityType;
-                var serializer = _entitySerializerCache.GetOrCreateSerializer(entityType);
+                var serializer = navigation.IsCollection
+                    ? new CollectionSerializationProvider().GetSerializer(navigation.ClrType)
```

Comment:
Why do we use a serializer cache for entities, but not for collection serializers?

Reply:
I'll look into unifying the cache for all types.